Microdata.js

```diff
@@ -0,0 +1,863 @@
+{
+	"translatorID": "fcb1b13c-afc8-453c-bd9c-399b06911e3a",
+	"label": "Microdata",
+	"creator": "Philipp Zumstein",
+	"target": "",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2016-05-29 19:17:42"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2016 Philipp Zumstein
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function extractMicrodata(doc, url) {
+	// The native DOM function will be removed soon:
+	// https://bugzilla.mozilla.org/show_bug.cgi?id=909633
+	// Thus we use here out own scraping methods for
+	// microdata.
+	
+	// Helpful info: https://blog.scrapinghub.com/2014/06/18/extracting-schema-org-microdata-using-scrapy-selectors-and-xpath/
+	
+	var schemaItems = ZU.xpath(doc, '//*[@itemscope]');
+	
+	// Assign the itemid to each item first
+	var usedTypes = [];
+	for (var i=0; i<schemaItems.length; i++) {
+		schemaItems[i].itemid = schemaItems[i].getAttribute("itemid") ||
+			(schemaItems[i].getAttribute("id") ? url+"#"+schemaItems[i].getAttribute("id") : url+"#itemid="+i);
```

Comment:
This is just to have some (meaningful) `itemid` also when the attribute is missing. I thought that the fragment identifier might be a good idea to have something which is unique.